internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -174,16 +178,33 @@ func newEnrollCmd(
 	options *enrollCmdOption,
 	configPath string,
 	store saver,
+	getDaemonState agentStateGetterFunc,
+	waitForAgentState agentStateGetterFunc,
+	hr func() (bool, error),
 ) (*enrollCmd, error) {
 	return &enrollCmd{
 		log:              log,
 		options:          options,
 		configStore:      store,
 		configPath:       configPath,
 		daemonReloadFunc: daemonReload,
+		getAgentStateWB:  agentStateBackoffFunc(getDaemonState, waitForAgentState, daemonTimeout),
+		hasRoot:          hr,
 	}, nil
 }
 
+var unprivilegedAgentRootUserError = errors.New("cannot execute this command as root, agent is unprivileged, use \"sudo -u elastic-agent-user enroll\"")
```

Comment:
See below.

But if you don't change the way this works then `sudo -u elastic-agent-user` cannot be hard-coded because the username will be able to be customized. @michalpristas Is adding that ability.

This also doesn't work on Windows, you would need to provide an error message that world work on Windows.